core/src/main/scala/spinal/core/Trait.scala

```diff
@@ -810,6 +810,8 @@ object noLatchCheck                  extends SpinalTag
 object noBackendCombMerge            extends SpinalTag
 object crossClockDomain              extends SpinalTag{ override def moveToSyncNode = true }
 object crossClockBuffer              extends SpinalTag{ override def moveToSyncNode = true }
+case class crossClockFalsePath(source: Option[BaseType] = None, destIsReset: Boolean = false) extends SpinalTag { override def allowMultipleInstance: Boolean = false }
+case class crossClockMaxDelay(cycles: Int, useTargetClock: Boolean) extends SpinalTag { override def allowMultipleInstance: Boolean = false }
```

Comment:
I see 3 cases here : 
- Using target clock (useTargetClock=true)
- Using source clock (?)
- Using the fastest of both clock to avoid race condition (?)


Also, maybe we should see it more named as a "multiCyclePath" ?

Reply:
I don't really see the "multiCyclePath" here: that's usually used for a signal path staying in its own clock domain, but allowing more then 1 cycle propagation delay. 
Here we are talking about a max. propagation delay (name is taken from the STA tools)

Reply:
Ahh good then :)

Reply:
Yeah as of now we can specify to use either the source or dest clock, but not the fastest.  Should this be done in Spinal (look at the CD and find out which one is faster), or in Tcl?

Reply:
Hmmm, i would say, keeping it in SpinalHDL is fine

Reply:
Then we can simply ask the caller to determine the appropriate clock domain to use.  Do we actually have a case where we need to use the fastest?

Reply:
As long as it isn't to the user to figure it out. I think that is ok.
But the one thing is that not everybody provide clock informations to SpinalHDL, so if the TCL generation isn't used, it should be fine not have clock domain frequancy specified

Reply:
Yeah I think as is it's good then?  It's an attribute pushed by the CDC library

Reply:
What is good ? 

> Do we actually have a case where we need to use the fastest?

Hmmm i'm not sure anymore.
- For a gray counter, you want the source clock for constraint.
- For sampling some data when you know it is stable, you want the destination clock fo constraint

But i may miss some special cases ?

Reply:
I'd somewhat argue against binding this to SpinalHDL timings here:
- as you already said, often we want source/destination clock for the constraint - independent of their speed
- in such cases we shouldn't require timings to be available - especially for the use-case of building reusable components
- if we really come across another use-case where we need "fastest" we can still add it

Reply:
Yeah the safest way is to do it in Tcl IMO, and that's how we implement it right now (apart from the clock defs, but that's of less concern).  Do we need any changes here?

Reply:
I think we are in agreement to keep it as-is, what about you @Dolu1990 ?

Reply:
> But i may miss some special cases ?

I think the reason why we used "fastest of both side" as constraints, was to be very safe by using a "design blind"  aproache. Meanwhile, if more care was used, we could have used the correct side of the CC as constraint.

=> Ok for me as-is.